### ISSUE TITLE
Temporary relations are left vulnerable to garbage collection

### ISSUE THREAD

**Issue by hazzabeee** [edbmods/EdBPrepareCarefully#314]
EdBPrepareCarefully leaves temporary relations (specifically the "placeholder" parent pawns used to create sibling relationships) vulnerable to garbage collection. That is, it does not give them any attributes which would mark them for keeping during GC (see RimWorld.Planet.WorldPawnGC) -- at least, no attributes that persist during a save/reload cycle. This is a problem because when these parents are GC'ed in game (typically after 48 hours), this destroys the sibling relationship.

Vanilla seems to "protect" its parent pawns by adding them to pawnsForcefullyKeptAsWorldPawns (during GameInitData.PrepForMapGen?), but EdBPrepareCarefully does not. (An alternative method would be to set the TimeAsColonistOrColonyAnimal record to any integer greater than zero.)

This appears to new behaviour in 1.3, possibly due to changes to RimWorld.Planet.WorldPawnGC.AccumulatePawnGCData, or (working hypothesis) because since commit e8057e50 EdBPrepareCarefully adds dead relations to the world as follows:
```
            // Killing a pawn adds it to the world
            if (pawn.Type == CustomPawnType.Temporary) {
                if (!pawn.Pawn.Dead) {
                    pawn.Pawn.Kill(null, null);
                }
                return;
            }
```

pawn.Pawn.kill in turn calls worldPawns.PassToWorld(this, PawnDiscardDecideMode.Decide), which does not add the pawn to pawnsForcefullyKeptAsWorldPawns. 

A tentative solution might be to manually call Find.WorldPawns.PassToWorld(pawn.Pawn, PawnDiscardDecideMode.KeepForever) before pawn.Pawn.Kill (and likewise in RelationshipBuilder). If I have time I can investigate further and see if I can get a patch together.

p.s. great mod, thanks fur putting it together :)

**To reproduce:**

- Create siblings using PrepareCarefully (with temporary/no parents)
- Load world and verify that they are siblings
- Can also use dev mode to verify that their dead parent(s) exist in the world
- Save and reload
- Using debug log to trigger garbage collection 

**Expected result:**

- Siblings remain siblings; dead parent(s) continue to exist in world

**Actual result:**

- Siblings cease to be siblings because their dead parent(s) are garbage collected.

**Comment by hazzabeee** [comment]
Well, after all of that something's updated and I now can't reproduce this issue in a new save file. So I guess the above supposition is all wrong and it already got fixed at some point!

**Comment by hazzabeee** [comment]
Ah, got it. You have to save and load first. I'll update the description.

**Comment by edbmods** [comment]
Thanks for reporting this.  I'll take a look at your commit and try to fix the potential issue for hidden pawns as well.

**Comment by edbmods** [comment]
Fixed in version 1.3.9
